inginious/frontend/pages/utils.py

```diff
@@ -151,7 +168,9 @@ def GET(self, *args, **kwargs):
         Otherwise, returns the login template.
         """
         if self.user_manager.session_logged_in():
-            if not self.user_manager.session_username() and not self.__class__.__name__ == "ProfilePage":
+            user = self.database.users.find_one({"email": self.user_manager.session_email()})
```

Comment:
Even if there many places in the code where this is already the case, it would be better to avoid requesting the user again from database (this method is checked for every page). Two solutions:
- Either we keep this piece of information in the session object and define a new function ``session_tos_accepted``
- Either we keep the complete user profile object in the session.

Another lightweight solution to explore would be to set the username to ``''`` at login if the TOS are not accepted, mimicking the behaviour of a non-set username.

Reply:
So this line should be useless now.